coursera_autograder/commands/resources.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+# Copyright 2015 Coursera
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Coursera's asynchronous grader command line SDK.
+
+You may install it from source, or via pip.
+"""
+
+from coursera_autograder.commands import common
+from coursera_autograder.commands import oauth2
+import logging
+import requests
+import urllib.parse
+
+"""
+target fields:
+  reservedCpu: int?
+  reservedMemory: int?
+  wallClockTimeout: int?
+"""
+
+def command_resources(args):
+    "Implements the resources subcommand"
+
+    oauth2_instance = oauth2.build_oauth2(args)
+    auth = oauth2_instance.build_authorizer()
+
+    s = requests.Session()
+    s.auth = auth
+
+    course_branch_id = args.course.replace("~", "!~") if "authoringBranch" in args.course else args.course
+    course_branch_item = '%s~%s' % (course_branch_id, args.item)
+
+    params = 'id=%s&partId=%s' % (course_branch_item, args.part)
+    result = s.post(args.executorInfo_endpoint, params=params)
+    if result.status_code == 404:
+        logging.error(
+            '\nUnable to find executor with part id %s in item %s in course %s.\n'
+            'Status Code: 404 \nURL: %s \nResponse: %s\n',
+            args.part, 
+            args.item, 
+            args.course,
+            result.url,
+            result.text)
+        return 1
+    elif result.status_code != 200:
+        logging.error(
+            '\nUnable to get executor resources.\n'
+            'CourseId: %s\n'
+            'ItemId: %s\n'
+            'PartId: %s\n'
+            'Status Code: %d \nURL: %s \nResponse: %s\n',
+            args.course,
+            args.item,
+            args.part,
+            result.status_code,
+            result.url,
+            result.text
+        )
+        return 1
+    print(
+        '\nResource Limits for executor with part id %s in item %s in course %s:\n'
+        'Reserved CPU (AWS units -- 1024 units = 1 vCPU): %s (%s)\n'
+        'Reserved Memory (MB): %s\n'
+        'Wall Clock Timeout (s): %s\n' %
+        (args.part,
+         args.item,
+         args.course,
+         result.json()['reservedCpu'],
+         int(result.json()['reservedCpu'])/1024,
+         result.json()['reservedMemory'],
+         result.json()['wallClockTimeout'] if 'wallClockTimeout' in result.json() else 'Timeout not set - default is 1200 seconds'))
```

Comment:
we need to do the same for Cpu and Memory, if the field is not set, show the default values.

Reply:
Done, but I don't think Cpu and Memory should ever be in a state where they aren't set.